apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Trace/TraceTest.cs

```diff
@@ -148,6 +149,44 @@ public async Task Trace_Header()
             }
         }
 
+        [Fact]
+        public async Task Trace_MultipleHeaderPropagation()
+        {
+            var createFirstSpanUri = $"/Propagation/{nameof(PropagationController.CreateFirstSpan)}/{_testId}";
+            var createFirstSpanSpanName = EntryData.GetMessage(nameof(PropagationController.CreateFirstSpan), _testId);
+
+            var createSecondSpanUri = $"/Propagation/{nameof(PropagationController.CreatesSecondSpan)}/{_testId}";
+            var createSecondSpanSpanName = EntryData.GetMessage(nameof(PropagationController.CreatesSecondSpan), _testId);
+
+            var createNoSpanUri = $"/Propagation/{nameof(PropagationController.CreateNoSpan)}/{_testId}";
+
+            using (var server = new TestServer(new WebHostBuilder().UseStartup<TraceTestNoBufferHighQpsApplication>()))
+            using (var client = server.CreateClient())
+            {
+                var response = await client.GetAsync(createFirstSpanUri);
+            }
+
+            var trace = _polling.GetTrace(createFirstSpanUri, _startTime);
+
+            Assert.NotNull(trace);
+
+            var parentSpan = trace.Spans.Single(s => s.Name == createFirstSpanUri);
```

Comment:
After a few minutes, I'm getting lost here. It might be simpler to have a separate variable for each span - and definitely a comment.

Reply:
My bad, I needed a diagram to write the test so no wonder you are having a hard time. Added comments and different variables.